src/base/http/connection.h

```diff
@@ -27,41 +28,69 @@
  * exception statement from your version.
  */
 
-
 #pragma once
 
 #include <QElapsedTimer>
 #include <QObject>
 
+#include "types.h"
+
 class QTcpSocket;
 
 namespace Http
 {
-    class IRequestHandler;
-    struct Response;
-
     class Connection : public QObject
     {
         Q_OBJECT
         Q_DISABLE_COPY(Connection)
 
     public:
-        Connection(QTcpSocket *socket, IRequestHandler *requestHandler, QObject *parent = nullptr);
-        ~Connection();
+        enum Status
```

Comment:
IMO I dislike having such state-machine. The http header fields/protocol are well defined and don't need state transitions. Can't you make it work without it?

Reply:
They were reason helpful debugging. I can make this enum private.